### PR TITLE
lib: location: Skyhook support removed

### PR DESCRIPTION
**Pull request by jhirsi**
Skyhook support removed from all related location related libraries, samples and documentation.

### CODE REVIEW

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -497,6 +497,16 @@ Modem libraries
       * Removed dependency on the :ref:`lib_modem_jwt` library.
         The :ref:`lib_location` library now selects :kconfig:option:`CONFIG_NRF_CLOUD_REST_AUTOGEN_JWT` when using :kconfig:option:`CONFIG_NRF_CLOUD_REST`.
 
+    * Removed:
+
+       * Skyhook support removed.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:502]:
```suggestion
       * Support for Skyhook.
```
The bullet above already suggests that something is removed.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -497,6 +497,16 @@ Modem libraries
...
+
+  * :ref:`lib_multicell_location` library:
+
+    * Removed:
+
+        * Skyhook support removed.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:508]:
```suggestion
        * Support for Skyhook.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (jhirsi) [reply doc/nrf/releases/release-notes-changelog.rst]: True. Changed

**Review by peknis — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by tokangas — APPROVED** [review]

**Comment by jhirsi** [comment]
Rebased again and hoping CI to pass.

**Comment by jhirsi** [comment]
Rebased again...

**Review by hiltunent — APPROVED** [review]

**Pull request merged**